projects/dotcms-field-elements/src/components/dot-textfield/dot-texfield.e2e.ts

```diff
@@ -1,140 +1,345 @@
+            });
+        });
+
+        describe('requiredMessage', () => {
+            it('should show default value of requiredMessage', async () => {
+                element.setProperty('required', 'true');
+                await input.press('a');
+                await input.press('Backspace');
+                await page.waitForChanges();
+                expect((await dotTestUtil.getErrorMessage(page)).innerText).toBe(
+                    'This field is required'
+                );
+            });
+
+            it('should show requiredMessage', async () => {
+                element.setProperty('required', 'true');
+                element.setProperty('requiredMessage', 'Test');
+                await input.press('a');
+                await input.press('Backspace');
+                await page.waitForChanges();
+                expect((await dotTestUtil.getErrorMessage(page)).innerText).toBe('Test');
+            });
+
+            it('should not render requiredMessage', async () => {
+                await page.waitForChanges();
+                expect(await dotTestUtil.getErrorMessage(page)).toBe(null);
+            });
+
+            it('should not render and not break with with invalid value', async () => {
+                element.setProperty('required', 'true');
+                element.setProperty('requiredMessage', { test: 'hi' });
+                await input.press('a');
+                await input.press('Backspace');
+                await page.waitForChanges();
+                expect(await dotTestUtil.getErrorMessage(page)).toBeNull();
+            });
+        });
+
+        describe('regexCheck', () => {
+            it('should set correct value when valid regexCheck', async () => {
+                element.setAttribute('regex-check', '[0-9]*');
+                await page.waitForChanges();
+                expect(await element.getProperty('regexCheck')).toBe('[0-9]*');
+            });
+
+            it('should set empty value when invalid regexCheck', async () => {
+                element.setAttribute('regex-check', '[*');
+                await page.waitForChanges();
+                expect(await element.getProperty('regexCheck')).toBe('');
+            });
+        });
+
+        describe('validationMessage', () => {
+            it('should render validationMessage', async () => {
+                element.setProperty('regexCheck', '[0-9]');
+                element.setProperty('validationMessage', 'Test');
+                await input.press('a');
+                await page.waitForChanges();
+                expect((await dotTestUtil.getErrorMessage(page)).innerText).toBe('Test');
+            });
+
+            it('should not render validationMessage', async () => {
+                await input.press('a');
+                await page.waitForChanges();
+                expect(await dotTestUtil.getErrorMessage(page)).toBeNull();
+            });
+        });
```

Comment:
should we do a test when required and regexCheck are both unvalid?

Reply:
required can't be invalid, invalid means `true` and that is tested in the `required` section, so I don't think is needed